webapp/app/dashboards/routes.py

```diff
@@ -634,54 +674,116 @@ def add_mean_over_sensors(sensor_ids, df):
     # "phase shifted" with respect to each other, e.g. one may have data for 00 and 10,
     # while another may have 05 and 15. A 10 minute rolling mean smooths out these
     # differences.
-    roll_window = timedelta(minutes=10)
-    for column_name in ("temperature", "humidity"):
+    roll_window = timedelta(minutes=roll_window_minutes)
+    for column in DATA_COLUMNS_BY_SENSOR_TYPE[sensor_type]:
+        column_name = column["column_name"]
         df_mean[column_name] = df_mean[column_name].rolling(roll_window).mean()
     df_mean = df_mean.reset_index()
     df = pd.concat((df_mean, df), axis=0)
     return df
 
 
+def fetch_all_sensor_types(valid_sensor_types):
+    """Get all sensor types from the CROP database, for which we know how to render the
+    timeseries dashboard.
+
+    Arguments:
+        valid_sensor_types: A list of IDs of sensor types that we know how to fetch and
+        plot data for. Sensor types in the database that are not in this list are
+        discarded.
+    Returns:
+        List of dictionaries with keys "id" (int) and "sensor_type" (str).
+    """
+    query = db.session.query(
+        TypeClass.id,
+        TypeClass.sensor_type,
+    ).filter(TypeClass.id.in_(valid_sensor_types))
+    sensor_types = db.session.execute(query).fetchall()
+    sensor_types = query_result_to_array(sensor_types)
+    return sensor_types
+
+
+def fetch_all_sensors(sensor_type):
+    """Get all sensors of a given sensor type from the CROP database.
+
+    Arguments:
+        sensor_type: The database ID (primary key) of the sensor type.
+    Returns:
+        List of dictionaries with keys "id" (int) and "name" (str), sorted by "id".
+    """
+    query = db.session.query(
+        SensorClass.id,
+        SensorClass.name,
+    ).filter(SensorClass.type_id == sensor_type)
+    sensors = db.session.execute(query).fetchall()
+    sensors = query_result_to_array(sensors)
+    sensors = list(sorted(sensors, key=lambda x: x["id"]))
+    return sensors
+
+
 @blueprint.route("/timeseries_dashboard", methods=["GET", "POST"])
 @login_required
 def timeseries_dashboard():
     # Read query string
     dt_from = request.args.get("startDate")
     dt_to = request.args.get("endDate")
     sensor_ids = request.args.get("sensorIds")
-    if dt_from is None or dt_to is None or sensor_ids is None:
+    if sensor_ids is not None:
+        sensor_ids = tuple(map(int, re.split(r"[ ;,]+", sensor_ids.rstrip(" ,;"))))
```

Comment:
A very fair point. It just splits a comma-separated string of ints into a list of ints. Okay, it can actually handle space and semicolon separated ones too, but that's just left-over from the previous way of inputting the sensor_ids. Comment added.